src/domain/users/user-organizations.repository.integration.spec.ts

```
@@ -0,0 +1,1414 @@
+        user: member.generatedMaps[0],
+        address: getAddress(faker.finance.ethereumAddress()),
+      });
+      const org = await dbOrgRepo.insert({
+        name: orgName,
+        status: 'ACTIVE',
+      });
+      const orgId = org.generatedMaps[0].id;
+      await dbUserOrgRepo.insert({
+        user: owner.generatedMaps[0],
+        organization: org.generatedMaps[0],
+        role: 'MEMBER',
+        status: 'ACTIVE',
+      });
+      await dbUserOrgRepo.insert({
+        user: member.generatedMaps[0],
+        organization: org.generatedMaps[0],
+        role: 'MEMBER',
+        status: 'ACTIVE',
+      });
+
+      await expect(
+        userOrgRepo.updateRole({
+          authPayload: new AuthPayload(authPayloadDto),
+          orgId,
+          role: 'ADMIN',
+          userId: memberUserId,
+        }),
+      ).rejects.toThrow('No user organizations found.');
+    });
+
+    it('should throw an error if downgrading the last ACTIVE ADMIN', async () => {
+      const authPayloadDto = authPayloadDtoBuilder().build();
+      const orgName = faker.word.noun();
+      const user = await dbUserRepo.insert({
+        status: 'ACTIVE',
+      });
+      const userId = user.generatedMaps[0].id;
+      await dbWalletRepo.insert({
+        user: user.generatedMaps[0],
+        address: authPayloadDto.signer_address,
+      });
+      const org = await dbOrgRepo.insert({
+        name: orgName,
+        status: 'ACTIVE',
+      });
+      const orgId = org.generatedMaps[0].id;
+      await dbUserOrgRepo.insert({
+        user: user.generatedMaps[0],
+        organization: org.generatedMaps[0],
+        role: 'ADMIN',
+        status: 'ACTIVE',
+      });
+
+      await expect(
+        userOrgRepo.updateRole({
+          authPayload: new AuthPayload(authPayloadDto),
+          orgId,
+          role: 'MEMBER',
+          userId,
+        }),
+      ).rejects.toThrow('Cannot remove last admin.');
+    });
+  });
+
+  describe('removeUser', () => {
```

Since Authentication/Authorization logic happens in the repo, we should add some tests to make sure an unauthorized user cannot remove members of an org.
The same applies to other methods like `updateRole`, `acceptInvite`, `declineInvite`, `inviteUser` but I won't add a comment on each method to keep the review concise.

---

I scaffolded new tests in d556d8e14bac3e535f72e269f55bb37a7031d792 that I will complete shortly.

---

Added the tests in df962dc640c99b5d2584271692f502637bff28e0.

---

`inviteUsers` is still missing `Authorization` tests.
nit: `updateRole`:
`Should throw an error if user does not have access to upgrade other members' roles`

---

I added relevant tests in cf92c90a239f7c658457c7765f2076c1f25bd9db.